cli/src/cmd/forge/script.rs

```diff
@@ -240,203 +146,93 @@ impl Cmd for ScriptArgs {
-                                Ok(Some(res)) => {
-                                    let tx_str = serde_json::to_string_pretty(&res).expect("Bad serialization");
-                                    println!("{}", tx_str);
-                                    executed_transaction_receipts.push(res);
-                                }
+            println!("Gas used: {}", result.gas);
+            println!("== Logs ==");
+            let console_logs = decode_console_logs(&result.logs);
+            if !console_logs.is_empty() {
+                for log in console_logs {
+                    println!("  {}", log);
+                }
+            }
 
-                                Ok(None) => {
-                                    panic!("Failed to get transaction receipt?")
-                                }
-                                Err(e) => panic!("Aborting! A transaction failed to send: {:#?}", e)
-                            };
-                        });
-
-                    let mut out = config.out.clone();
-                    let target_fname = target.source.file_name().expect("No file name");
-                    out.push(target_fname);
-                    out.push("scripted_transactions");
-                    std::fs::create_dir_all(out.clone())?;
-                    out.push(self.sig.clone() + &format!("_executed_{}.json", chain));
-                    let mut file = std::fs::File::create(out.clone())?;
-                    file.write_all(
-                        serde_json::to_string_pretty(&executed_transaction_receipts)
-                            .expect("Bad serialization of receipts")
-                            .as_bytes(),
-                    )?;
+            println!("==========================");
+            println!("Simulated On-chain Traces:\n");
+            if let Some(txs) = result.transactions {
+                if let Ok(gas_filled_txs) =
+                    self.execute_transactions(txs, &evm_opts, &config, &mut decoder)
+                {
                     println!("\n\n==========================");
-                    println!("\nONCHAIN EXECUTION COMPLETE & SUCCESSFUL. Transaction receipts written to {:?}", out);
+                    if !result.success {
+                        panic!("\nSIMULATION FAILED");
+                    } else {
+                        let txs = gas_filled_txs;
+                        let mut deployment_sequence =
+                            ScriptSequence::new(txs, &self.sig, &target, &config.out)?;
+                        deployment_sequence.save()?;
+
+                        if self.execute {
+                            self.send_transactions(&mut deployment_sequence)?;
+                        } else {
+                            println!("\nSIMULATION COMPLETE. To send these transaction onchain, add `--execute` & wallet configuration(s) to the previously ran command. See forge script --help for more.");
```

Comment:
nit: minor wording and consistency between using ticks for flags or not

```suggestion
                            println!("\nSIMULATION COMPLETE. To broadcast these transactions, add --execute and wallet configuration(s) to the previous command. See forge script --help for more.");
```